packages/betterauth/src/plugins/usage.ts

```diff
@@ -0,0 +1,175 @@
+import {
+  APIError,
+  createAuthEndpoint,
+  sessionMiddleware,
+} from "better-auth/api";
+import type { DodoPayments } from "dodopayments";
+import {
+  UsageEventIngestResponse,
+  EventsDefaultPageNumberPagination,
+} from "dodopayments/resources/usage-events.mjs";
+import { z } from "zod/v3";
+
+const EventInputSchema = z.object({
+  event_id: z.string(),
+  event_name: z.string(),
+  metadata: z
+    .record(z.union([z.string(), z.number(), z.boolean()]))
+    .nullable()
+    .optional(),
+  timestamp: z.string().date().optional(),
+});
+
+export const usage = () => (dodopayments: DodoPayments) => {
+  return {
+    // Ingest usage data
+    dodoUsageIngest: createAuthEndpoint(
+      "/dodopayments/usage/ingest",
+      {
+        method: "POST",
+        use: [sessionMiddleware],
+      },
+      async (ctx): Promise<UsageEventIngestResponse> => {
+        if (!ctx.context.session?.user?.id) {
+          throw new APIError("BAD_REQUEST", {
+            message: "User not found",
+          });
+        }
+
+        if (!ctx.context.session?.user.emailVerified) {
+          throw new APIError("UNAUTHORIZED", {
+            message: "User email not verified",
+          });
+        }
+
+        const body = EventInputSchema.safeParse(ctx.body);
```

Comment:
@deveshru2712 why are we doing a `safeParse` at runtime when we could specify a `body` zod schema in the second argument for the `createAuthEndpoint`?